pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -16,6 +16,14 @@ showSyntax() {
   exit 1
 }
 
+sudo() {
+  if command sudo --help | grep -- --preserve-env= >/dev/null; then
+    command sudo -H --preserve-env=PATH "$@"
```

Comment:
```suggestion
    command sudo -H --preserve-env=PATH env "$@"
```

As the `env` command was added to ensure that the preserved `PATH` gets searched for the binary, I think it'd be better to move it here